src/AppBundle/Controller/Report/ProfDeputyPrevCostController.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace AppBundle\Controller\Report;
+
+use AppBundle\Controller\RestController;
+use AppBundle\Entity as EntityDir;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\HttpFoundation\Request;
+
+class ProfDeputyPrevCostController extends RestController
+{
+    private $sectionIds = [EntityDir\Report\Report::SECTION_PROF_DEPUTY_COSTS];
+
+    /**
+     *
+     * @Route("/report/{reportId}/prof-deputy-previous-cost")
+     * @Method({"POST"})
+     * @Security("has_role('ROLE_PROF')")
+     */
+    public function addAction(Request $request, $reportId)
+    {
+        $data = $this->deserializeBodyContent($request);
+
+        /* @var $report EntityDir\Report\Report */
+        $report = $this->findEntityBy(EntityDir\Report\Report::class, $reportId);
+        $this->denyAccessIfReportDoesNotBelongToUser($report);
+        if (!array_key_exists('amount', $data)) {
+            throw new \InvalidArgumentException("Missing amount");
+        }
+        $cost = new EntityDir\Report\ProfDeputyPreviousCost($report, $data['amount']);
+        $this->updateEntity($data, $cost);
+        $report->setProfDeputyCostsHasPrevious('yes');
+        $this->persistAndFlush($cost);
+
+        $report->updateSectionsStatusCache($this->sectionIds);
+        $this->getEntityManager()->flush();
```

Comment:
Not sure the cache status should be updated before the data is flushed...?

Reply:
I don't think it matters because the `updateSectionsStatusCache` does not read the latest state from the database, but from the in memory entity object instead.

Reply:
ok I wasn't sure, will leave as is